File: pact-jvm-provider/src/main/java/au/com/dius/pact/provider/junit/Provider.java
package au.com.dius.pact.provider.junit;

import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Used to pass provider name to Pact runner
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@Inherited
public @interface Provider {
    /**
     * @return provider name for pact test running
     */
    String value();
}

File: provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/Provider.kt
package au.com.dius.pact.provider.junitsupport

import java.lang.annotation.Inherited
import kotlin.annotation.Retention

/**
 * Used to pass provider name to Pact runner. Can use expressions (in `${}` form) to get the value from Java system
 * properties or environment variables.
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
@Inherited
annotation class Provider(
  /**
   * @return provider name for pact test running
   */
  val value: String = ""
)
